engine/src/execution_graph/logic_controllers/BatchProcessing.h

```diff
@@ -273,12 +298,38 @@ class BindableTableScan : public PhysicalPlan {
 			try {
 				auto batch = input.next();
 
-				if(is_filtered_bindable_scan(expression)) {
-					auto columns = ral::processor::process_filter(batch->toBlazingTableView(), expression, context.get());
-					this->add_to_output_cache(std::move(columns));
-				}
-				else{
-					this->add_to_output_cache(std::move(batch));
+				if (is_filtered_bindable_scan(expression)) {
+					// For large files
+					if (batch->sizeInBytes() > 1073741824) {  // Threshold 1GB
+						std::size_t total_partitions = 10;    // TODO: Looking for the optimal value
+						std::vector<cudf::size_type> splits(total_partitions);
+						std::iota(splits.begin(), splits.end(), 0);
+
+						std::vector<CudfTableView> partitioned = cudf::experimental::split(batch->view(), splits);
+						for (std::size_t batch_i = 0; batch_i < partitioned.size(); ++batch_i) {
+							ral::frame::BlazingTableView batch_small(partitioned[batch_i], batch->names());
+							auto columns = ral::processor::process_filter(batch_small, expression, context.get());
+							this->add_to_output_cache(std::move(columns));
+						}
+					} else {
+						auto columns = ral::processor::process_filter(batch->toBlazingTableView(), expression, context.get());
+						this->add_to_output_cache(std::move(columns));
+					}
+				} else {
+					// For large files
+					if (batch->sizeInBytes() > 1073741824) {  // Threshold 1GB
```

Comment:
This is the third block of code where you take a batch and see if you want to break it up into smaller batches. You should make that a function so that you dont repeat yourself

Reply:
I will remove it!